### PR TITLE
[UP-247] GlobalSign DSS Cert and OCSP support

### PR DESCRIPTION
**Pull request by sampila**
Trying to implement `github.com/unidoc/timestamp`, but due `timestamp` package already importing this package, import cycle is not allowed.

### CODE REVIEW

File: sign.go [hunk 1]
```diff
@@ -339,6 +348,45 @@ func (si *signerInfo) SetUnauthenticatedAttributes(extraUnsignedAttrs []Attribut
 	return nil
 }
 
+type TimestampTokenRequestCallback func(digest []byte) ([]byte, error)
```
Comment (gunnsth) [inline sign.go:352]: Add comment on exported type

File: sign.go [hunk 2]
```diff
@@ -339,6 +348,45 @@ func (si *signerInfo) SetUnauthenticatedAttributes(extraUnsignedAttrs []Attribut
...
+
+func (sd *SignedData) RequestSignerTimestampToken(signerID int, callback TimestampTokenRequestCallback) error {
```
Comment (gunnsth) [inline sign.go:356]: Add comment on exported function

File: sign.go [hunk 3]
```diff
@@ -339,6 +348,45 @@ func (si *signerInfo) SetUnauthenticatedAttributes(extraUnsignedAttrs []Attribut
...
+
+	tst, err := callback(sd.sd.SignerInfos[signerID].EncryptedDigest)
+	if err != nil {
+		return err
+	}
+	return sd.AddTimestampTokenToSigner(signerID, tst)
+}
+
+// AddTimestampTokenToSigner inserts TimestampToken which described in RFC3161 into
+// unauthenticated attribute of that signer.
```
Comment (gunnsth) [inline sign.go]: define what `signerID` is ? and `tst` ?

File: sign.go (hunk 1 above)
Comment (sampila) [reply sign.go:352]: Updated at 3807938c83f951ee5425c3fc528f87543537f443

File: sign.go (hunk 2 above)
Comment (sampila) [reply sign.go:356]: Updated at 3807938c83f951ee5425c3fc528f87543537f443

File: sign.go (hunk 3 above)
Comment (sampila) [reply sign.go]: Updated at 3807938c83f951ee5425c3fc528f87543537f443

File: sign.go [hunk 4]
```diff
@@ -339,6 +348,48 @@ func (si *signerInfo) SetUnauthenticatedAttributes(extraUnsignedAttrs []Attribut
...
+
+	tst, err := callback(sd.sd.SignerInfos[signerID].EncryptedDigest)
+	if err != nil {
+		return err
+	}
+	return sd.AddTimestampTokenToSigner(signerID, tst)
+}
+
+// AddTimestampTokenToSigner inserts `tst` TimestampToken which described in RFC3161 into
+// unauthenticated attribute of `signerID`.
```
Comment (gunnsth) [inline sign.go]: It seems like the ID is an index? Might be useful to mention that? Or is this terminology used in this context?

Comment (sampila) [reply sign.go]: Updated at 12f70a22a9160d9d86634d72e04dcc7d023f1a9a

**Review by adrg — APPROVED** [review]
LGTM.

**Pull request merged**